src/cascadia/PublicTerminalCore/HwndTerminal.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/dx/DxRenderer.hpp"
+#include "../../cascadia/TerminalCore/Terminal.hpp"
+
+extern "C" {
+    __declspec(dllexport) HRESULT _stdcall CreateTerminal(HWND parentHwnd, _Out_ void** hwnd, _Out_ void** terminal);

```

Comment:
This is really gross and is screaming "IDL" and/or "COM" at me. Maybe `TerminalCore` just needs to become a CppWinRT module and be activated that way?

Or can we not do that because you're trying to make this work back to Windows 7? In which case, can it just be a classic COM interface definition or something?

Reply:
Or god forbid... C++/CX.

Reply:
I did it this way since WPF has to be done via managed code. I don't see how rewriting TerminalCore to CppWinRT would fix that problem. I could have done registration-free COM but C-API via PInvoke seemed more straightforward.

Reply:
Alright, that's fine.